src/handlers/mozilla-auth0.js

```diff
@@ -0,0 +1,109 @@
+import User from './../user'
+import assert from 'assert'
+import _ from 'lodash';
+import jwt from 'jsonwebtoken';
+import expressJwt from 'express-jwt';
+import jwks from 'jwks-rsa';
+import Debug from 'debug';
+import auth0js from 'auth0-js';
+
+const request = require('superagent-promise')(require('superagent'), Promise);
+
+const debug = Debug('handlers.mozilla-auth0');
+
+class Handler {
+  constructor({name, cfg}) {
+    let handlerCfg = cfg.handlers[name];
+    assert(handlerCfg.domain, `${name}.domain is required`);
+    assert(handlerCfg.apiAudience, `${name}.apiAudience is required`);
+    assert(handlerCfg.clientId, `${name}.clientId is required`);
+    assert(handlerCfg.clientSecret, `${name}.clientSecret is required`);
+    assert(handlerCfg.mgmtApiToken, `${name}.mgmtApiToken is required`);
+    _.assign(this, handlerCfg);
+
+    // use express-jwt to validate JWTs against auth0
+    this.jwtCheck = expressJwt({
+      secret: jwks.expressJwtSecret({
+        cache: true,
+        rateLimit: true,
+        jwksRequestsPerMinute: 5,
+        jwksUri: `https://${this.domain}/.well-known/jwks.json`,
+      }),
+      // expect to see our audience in the JWT
+      audience: this.apiAudience,
+      // and expect a token issued by auth0
+      issuer: `https://${this.domain}/`,
+      algorithms: ['RS256'],
+      credentialsRequired: true,
+    });
+
+    this._managementApiExp = null;
+    this._managementApi = null;
+  }
+
+  // Get a management API instance, by requesting an API token as needed
+  // see https://auth0.com/docs/api/management/v2/tokens
+  async getManagementApi() {
+    if (this._managementApi && new Date().getTime() / 1000 < this._managementApiExp - 10) {
+      return this._managementApi;
+    }
+
+    let res = await request.post(`https://${this.domain}/oauth/token`)
```

Comment:
I can't remember superagent-promise anymore... but make sure this does retries.
(maybe consider using `got`)